### PR TITLE
Localisation pr

### PR DESCRIPTION
**Pull request by mzeuner**
We define ocalisation of a commutative ring at a multiplicatively closed subset and prove that this is again a commutative ring.
We also look at the special case where single elements get inverted and start showing that inverting two elements successively is the same as inverting the product of the two elements. We also add some initial experiments regarding different ways of defining localisation.

This is a clean-up of the now closed pull request (https://github.com/agda/cubical/pull/399). Unfortunately the HIT version using instance arguments didn't survive. Maybe someone has an idea on how to bring them back into the picture...

### CODE REVIEW

File: Cubical/HITs/SetQuotients/Properties.agda [hunk 1]
```diff
@@ -128,6 +130,57 @@ setQuotUniversal Bset = isoToEquiv (iso intro out outRightInv outLeftInv)
 
 open BinaryRelation
 
+-- characterisation of binary functions/operations on set-quotients
+setQuotUniversal2 : {B : Type ℓ} (Bset : isSet B) → isRefl R
+                  → (A / R → A / R → B)
+                  ≃ (Σ[ _∗_ ∈ (A → A → B) ] ((a a' b b' : A) → R a a' → R b b' → a ∗ b ≡ a' ∗ b'))
```
Comment (mzeuner) [inline Cubical/HITs/SetQuotients/Properties.agda:136]: State as Iso AND equivalence

File: Cubical/Experiments/LocalisationDefs.agda [hunk 2]
```diff
@@ -0,0 +1,143 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
```
Comment (mzeuner) [inline Cubical/Experiments/LocalisationDefs.agda:1]: header!

File: Cubical/Algebra/CommRing/Localisation/Base.agda [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Powerset
+open import Cubical.Foundations.Transport
+open import Cubical.Functions.FunExtEquiv
+
+import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Bool
+open import Cubical.Data.Nat renaming (_+_ to _+ℕ_ ; _·_ to _·ℕ_ ; +-comm to +ℕ-comm ; +-assoc to +ℕ-assoc ; ·-assoc to ·ℕ-assoc ; ·-comm to ·ℕ-comm)
```
Comment (mzeuner) [inline Cubical/Algebra/CommRing/Localisation/Base.agda:16]: line break

File: Cubical/Algebra/CommRing/Localisation/Base.agda [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+private
+  variable
+    ℓ ℓ' : Level
+    A : Type ℓ
+
+
+-- A sub monoid of a commutative ring is a multiplicatively closed subset
+-- that contains 1
+record isSubMonoid (R' : CommRing {ℓ}) (S' : ℙ (R' .fst)) : Type ℓ where
```
Comment (mzeuner) [inline Cubical/Algebra/CommRing/Localisation/Base.agda:43]: isMultClosedSubset better?!

**Comment by mortberg** [comment]
Thanks for this PR!

**Pull request merged**